config/provider/inmem.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Layer5, Inc.
```

Comment:
@mgfeller  Does it make more sense to have the provider inside the config package? As it is just a config provider. We could have more different providers for tracing, cms etc. Correct me if I'm wrong.